pinot-core/src/main/java/org/apache/pinot/core/query/postaggregation/PostAggregationFunction.java

```diff
@@ -38,20 +37,20 @@ public class PostAggregationFunction {
 
   public PostAggregationFunction(String functionName, ColumnDataType[] argumentTypes) {
     String canonicalName = FunctionRegistry.canonicalize(functionName);
-    FunctionInfo functionInfo = FunctionRegistry.lookupFunctionInfo(canonicalName, argumentTypes);
+    int numArguments = argumentTypes.length;
+    FunctionInfo functionInfo = FunctionRegistry.lookupFunctionInfo(canonicalName, numArguments);
```

Comment:
Can we be sure this is not going to happen in other cases?

Reply:
This doesn't seem to be an issue in the v2 engine where Calcite will ensure the matching argument types during query compilation itself. And this `PostAggregationFunction` seems to be the only v1 engine specific piece that was updated in https://github.com/apache/pinot/pull/13573 to use the function lookup by argument type. In case we want to be extra safe though, we could do this - https://github.com/apache/pinot/pull/13711#issuecomment-2258300426.

Reply:
> In case we want to be extra safe though, we could do this - https://github.com/apache/pinot/pull/13711#issuecomment-2258300426.

To clarify, this will ensure that argument type based function lookup will function the same as the earlier argument count based function lookup in case of heterogeneous argument types in all cases (and we can un-revert the `PostAggregationFunction` changes). Actually, on thinking about this a bit more, I think we should definitely do so because even if we fix the literal handling in `PostAggregationHandler`, we'll still run into issues with `DOUBLE` / `INT` comparisons and the like in the v1 engine.